src/libpmem/pmem.c

```diff
@@ -1225,6 +1018,51 @@ pmem_get_cpuinfo(void)
 			Func_predrain_fence = predrain_fence_sfence;
 		}
 	}
+
+	char *ptr = os_getenv("PMEM_NO_MOVNT");
+	if (ptr && strcmp(ptr, "1") == 0) {
+		LOG(3, "PMEM_NO_MOVNT forced no movnt");
+	} else {
+#if SSE2_AVAILABLE
+		Func_memmove_nodrain = memmove_nodrain_sse2;
+		Func_memset_nodrain = memset_nodrain_sse2;
+#else
+		LOG(3, "sse2 disabled at build time");
+#endif
+
+		if (is_cpu_avx_present()) {
+#if AVX_AVAILABLE
+			LOG(3, "avx supported");
+
+			char *e = os_getenv("PMEM_NO_AVX");
```

Comment:
Rather than creating a bunch of independent PMEM_* environment variables, consider adding one PMEM_TUNABLES variable modeled after GLIBC_TUNABLES (https://www.gnu.org/software/libc/manual/html_node/Tunables.html).  Present a consistent user experience for controlling similar things.  Current glibc memcpy() related tunables include hwcaps (with flags including allowing AVX, AVX2, and AVX512 usage), x86_data_cache_size, x86_shared_cache_size, and x86_non_temporal_threshold.

Reply:
That's a very good idea. We have something similar in pmemobj, called [pmemobj_ctl_[set|get] / PMEMOBJ_CONF](http://pmem.io/pmdk/manpages/linux/master/libpmemobj/pmemobj_ctl_get.3). Porting it to libpmem is a few days of work (code, tests, documentation), which currently I can't spend time on. In order not to forget about it I created this feature request - pmem/issues#786.